x-pack/plugins/enterprise_search/public/applications/app_search/components/synonyms/synonyms.tsx

```diff
@@ -5,23 +5,86 @@
  * 2.0.
  */
 
-import React from 'react';
+import React, { useEffect } from 'react';
 
-import { EuiPageHeader, EuiPageContentBody } from '@elastic/eui';
+import { useValues, useActions } from 'kea';
+
+import {
+  EuiPageHeader,
+  EuiButton,
+  EuiPageContentBody,
+  EuiSpacer,
+  EuiFlexGrid,
+  EuiFlexItem,
+  EuiPagination,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
 
 import { FlashMessages } from '../../../shared/flash_messages';
 import { SetAppSearchChrome as SetPageChrome } from '../../../shared/kibana_chrome';
+import { Loading } from '../../../shared/loading';
 import { getEngineBreadcrumbs } from '../engine';
 
+import { SynonymCard, EmptyState } from './components';
 import { SYNONYMS_TITLE } from './constants';
 
+import { SynonymsLogic } from './';
+
 export const Synonyms: React.FC = () => {
+  const { loadSynonyms, onPaginate } = useActions(SynonymsLogic);
+  const { synonymSets, meta, dataLoading } = useValues(SynonymsLogic);
+  const hasSynonyms = synonymSets.length > 0;
+
+  useEffect(() => {
+    loadSynonyms();
+  }, [meta.page.current]);
+
+  useEffect(() => {
+    // If users delete the only synonym set on the page, send them back to the previous page
+    if (!hasSynonyms && meta.page.current !== 1) {
+      onPaginate(meta.page.current - 1);
+    }
+  }, [synonymSets]);
```

Comment:
Here's a gif of this behavior, hopefully it helps explain it:

![deleting-pagination](https://user-images.githubusercontent.com/549407/115811375-6dfa0d00-a3a4-11eb-98ef-26d1ead78074.gif)

Without this effect, the user ends up stranded on an empty page if they delete everything on that page.

Fun fact: this UX comes for free with EuiBasicTable's pagination, so we haven't had to deal with it before with other views, but sadly it doesn't come for free with just EuiPagination 🎺